translation-web/src/main/java/eu/europeana/api/translation/web/service/LangDetectionWebService.java

```diff
@@ -33,17 +35,20 @@ public LangDetectResponse detectLang(LangDetectRequest langDetectRequest)
     LanguageDetectionService fallback = getFallbackService(langDetectRequest);
     List<String> langs = null;
     String serviceId = null;
+    List<String> eligibleValues = new ArrayList<>();
     try {
+      // preprocess the values
+      eligibleValues =  translationServiceProvider.getLanguageDetectionPreProcessor().detectLang(langDetectRequest.getText(), langDetectRequest.getLang());
```

Comment:
this will have a side effect for the response in the case of exceptions, not sure if there are some cases where and exception will be thrown. Would be good to clarify with hugo and eventally leave a comment to handle exceptions if they occur